### DISCUSSION TITLE
[help🥹]how to distinguish different connections in a global context var

### DISCUSSION THREAD

**Discussion by vanker-x** [emmett-framework/emmett#496] [Q&A]
Hello, first of all, I am a python rookie. I am very interested in this project.
One of my doubts at present is how to distinguish different connections in a global ContextVar like websocket.
Because I read the code inside and did not understand it, I want to ask.
I am now very, very much looking forward to learning the technical details here.
The following is my hypothetical ASGI code of unimplemented details.
Assuming that the first connection is just blocked in the receive,
the second connection will replace the first connection when it enters.
How to distinguish the two connections is my doubt.
```python

class Websocket:
    def __init__(self, scope, receive, send):
        pass

    async def accept(self):
        pass

    async def receive(self):
        pass

    async def send(self, message):
        pass

    async def close(self):
        pass


from contextvars import ContextVar

not_bound = object()
websocket: ContextVar[Websocket] = ContextVar('websocket', default=not_bound)


async def asgi_app(scope, receive, send):
    token = websocket.set(Websocket(scope, receive, send))
    await websocket.get().accept()
    while True:
        message = await websocket.get().receive()
        if not message:
            break
    await websocket.get().close()
    websocket.reset(token)

```

**Comment by gi0baro** [comment]
That's exactly what contextvars are made for.
Not sure why you asking this in Emmett's discussions, sounds like you want to read [Python documentation](https://docs.python.org/3/library/contextvars.html) about that end eventually [the PEP](https://peps.python.org/pep-0567/).
